pkg/scorecard/scorecard_test.go

```diff
@@ -365,56 +364,100 @@ func Test_findConfigFile(t *testing.T) {
 			found:     "scorecard.yml",
 			wantFound: true,
 		},
+		{
+			desc:      "scorecard.yaml exists",
+			locs:      []string{"scorecard.yaml"},
+			found:     "scorecard.yaml",
+			wantFound: true,
+		},
 		{
 			desc:      ".scorecard.yml exists",
 			locs:      []string{".scorecard.yml"},
 			found:     ".scorecard.yml",
 			wantFound: true,
 		},
+		{
+			desc:      ".scorecard.yaml exists",
+			locs:      []string{".scorecard.yaml"},
+			found:     ".scorecard.yaml",
+			wantFound: true,
+		},
 		{
 			desc:      ".github/scorecard.yml exists",
 			locs:      []string{".github/scorecard.yml"},
 			found:     ".github/scorecard.yml",
 			wantFound: true,
 		},
+		{
+			desc:      ".github/scorecard.yaml exists",
+			locs:      []string{".github/scorecard.yaml"},
+			found:     ".github/scorecard.yaml",
+			wantFound: true,
+		},
 		{
 			desc:      "multiple configs exist",
-			locs:      []string{"scorecard.yml", ".github/scorecard.yml"},
+			locs:      []string{"scorecard.yml", ".github/scorecard.yaml"},
 			found:     "scorecard.yml",
 			wantFound: true,
 		},
 		{
-			desc:      "no config exists so shouldn't find one",
+			desc:      "no config exists",
 			locs:      []string{},
+			found:     "",
 			wantFound: false,
 		},
 	}
 
+	// Define the order of locations to check
+	checkOrder := []string{
+		"scorecard.yml",
+		"scorecard.yaml",
+		".scorecard.yml",
+		".scorecard.yaml",
+		".github/scorecard.yml",
+		".github/scorecard.yaml",
+	}
+
 	for _, tt := range tests {
 		tt := tt
 		t.Run(tt.desc, func(t *testing.T) {
 			t.Parallel()
 			ctrl := gomock.NewController(t)
 			mockRepoClient := mockrepo.NewMockRepoClient(ctrl)
-			mockRepoClient.EXPECT().GetFileReader(gomock.Any()).AnyTimes().DoAndReturn(func(filename string) (io.ReadCloser, error) {
-				if !slices.Contains(tt.locs, filename) {
-					return nil, fmt.Errorf("os.Open: %s", filename)
-				}
-				fullPath := filepath.Join("./testdata", filename)
-				f, err := os.Open(fullPath)
-				if err != nil {
-					return nil, fmt.Errorf("os.Open: %w", err)
-				}
-				return f, nil
-			})
-			r, path := findConfigFile(mockRepoClient)
+			defer ctrl.Finish()
 
-			if tt.found != "" && tt.found != path {
-				t.Errorf("expected config file %+v got %+v", tt.found, path)
+			// Set up expectations for each location in the correct order
+			for _, loc := range checkOrder {
+				if slices.Contains(tt.locs, loc) {
+					mockRepoClient.EXPECT().
+						GetFileReader(loc).
+						Return(io.NopCloser(strings.NewReader("test config")), nil)
+					// If this is the first matching file, we should stop here
+					if loc == tt.found {
+						break
+					}
```

Comment:
I don't think the mock client should be concerned with the check order or when it should stop. Just which files it should say exist or not based on `tt.locs`. Was there something wrong with the old setup?

Reply:
I don't believe this was resolved